bot/seasons/easter/egg_hunt/cog.py

```diff
@@ -0,0 +1,509 @@
+import asyncio
+import contextlib
+import logging
+import random
+import sqlite3
+from datetime import datetime, timezone
+from pathlib import Path
+
+import discord
+from discord.ext import commands
+
+from bot.constants import Channels, Client, Roles as MainRoles, bot
+from bot.decorators import with_role
+from .constants import Colours, EggHuntSettings, Emoji, Roles
+
+log = logging.getLogger(__name__)
+
+DB_PATH = Path("bot", "resources", "persist", "egg_hunt.sqlite")
+
+TEAM_MAP = {
+    Roles.white: Emoji.egg_white,
+    Roles.blurple: Emoji.egg_blurple,
+    Emoji.egg_white: Roles.white,
+    Emoji.egg_blurple: Roles.blurple
+}
+
+GUILD = bot.get_guild(Client.guild)
+
+
+def get_team_role(user: discord.Member):
+    """Helper function to get the team role for a member."""
+
+    if Roles.white in user.roles:
+        return Roles.white
+    if Roles.blurple in user.roles:
+        return Roles.blurple
+
+
+async def assign_team(user: discord.Member):
+    """Helper function to assign a new team role for a member."""
+    db = sqlite3.connect(DB_PATH)
+    c = db.cursor()
+    c.execute(f"SELECT team FROM user_scores WHERE user_id = {user.id}")
+    result = c.fetchone()
+    if not result:
+        new_team = random.choice([Roles.white, Roles.blurple])
+        log.debug(f"Assigned role {new_team} to {user}.")
+    else:
+        if result[0] == "WHITE":
+            new_team = Roles.white
+        else:
+            new_team = Roles.blurple
+        log.debug(f"Restored role {new_team} to {user}.")
+
+    await user.add_roles(new_team)
+    return GUILD.get_member(user.id)
+
+
+class EggMessage:
+    """Handles a single egg reaction drop session."""
+
+    def __init__(self, message: discord.Message, egg: discord.Emoji):
+        self.message = message
+        self.egg = egg
+        self.first = None
+        self.users = []
+        self.teams = {Roles.white: "WHITE", Roles.blurple: "BLURPLE"}
+        self.new_team_assignments = {}
+        self.timeout_task = None
+
+    @staticmethod
+    def add_user_score(user_id: int, team: str, score: int):
+        """Builds the SQL for adding a score to a user in the database."""
+
+        return (
+            "INSERT INTO user_scores(user_id, team, score)"
+            f"VALUES({user_id}, '{team}', {score})"
+            f"ON CONFLICT (user_id) DO UPDATE SET score=score+{score}"
+        )
+
+    @staticmethod
+    def add_team_score(team_name: str, score: int):
```

Comment:
same here. this method name feels misleading

Perhaps something like `add_team_score_sql` would be more precise.